### PR TITLE
Fix: Deprecated methods / Warnings / Imports

### PR DESCRIPTION
**Pull request by qmonmert**
- Update deprecated methods (java)
- Remove warnings (java)
- Remove unused imports (java)

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/test/java/package/web/rest/UserJWTControllerIT.java.ejs [hunk 1]
```diff
@@ -72,7 +72,8 @@ import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
 import static org.hamcrest.Matchers.nullValue;
-import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.Matchers.emptyString;
```
Comment (avdev4j) [inline generators/server/templates/src/test/java/package/web/rest/UserJWTControllerIT.java.ejs]: Does isEmptyString is deprecated ? Why should we use is(emptyString()) instead ?

Comment (qmonmert) [reply generators/server/templates/src/test/java/package/web/rest/UserJWTControllerIT.java.ejs]: yes it is deprecated http://hamcrest.org/JavaHamcrest/javadoc/2.0.0.0/org/hamcrest/text/IsEmptyString.html

Comment (avdev4j) [reply generators/server/templates/src/test/java/package/web/rest/UserJWTControllerIT.java.ejs]: oh great ! My bad, I had read the wrong javadoc version ;)

**Comment by avdev4j** [comment]
Let's merge this PR when all build goes green

**Comment by qmonmert** [comment]
@avdev4j you can merge :)

**Pull request merged**